lib/Evaluate/shape.cpp

```diff
@@ -316,6 +323,13 @@ MaybeExtentExpr GetUpperBound(
         }
       }
     }
+  } else if (const auto *assoc{
+                 symbol.detailsIf<semantics::AssocEntityDetails>()}) {
+    if (auto shape{GetShape(context, assoc->expr())}) {
+      if (!shape->empty()) {
+        return std::move(shape->back());
```

Comment:
dimension here too?

Does GetLowerBound also need similar code?

Reply:
Yes, and no (lower bounds of associate construct entities that aren't whole variables will always be 1).

Reply:
Well, that's not entirely true.  Associate construct entities that are associated with whole components can also have lower bounds other than 1.  And pointers, too.  I'll think more on this tomorrow.